File: python/minicurso/exemplos-java/src/pythonjava/Maioridade.java
package pythonjava;

class Pessoa {

	private String nome;
	private int idade;

	public Pessoa(String nome, int idade) {
		this.nome = nome;
		this.idade = idade;
	}

	public boolean maiorDeIdade() {
		return idade >= 18;
	}

	public String getNome() {
		return nome;
	}

}

public class Maioridade {

	public static void main(String[] args) {
		Pessoa[] pessoas = new Pessoa[] {
				new Pessoa("Joana", 29),
				new Pessoa("Joao", 34),
				new Pessoa("Joaozinho", 0),
		};

		System.out.println("Maiores de idade:");

		for (Pessoa p : pessoas) {
			if (p.maiorDeIdade()) {
				System.out.println(p.getNome());
			}
		}
	}
}

File: python/minicurso/exemplos_python/maioridade.py
class Pessoa:
    def __init__(self, nome, idade):
        self._nome = nome
        self._idade = idade

    def maior_de_idade(self):
        return self._idade >= 18

    def nome(self):
        return self._nome

pessoas = [
    Pessoa('Joana', 29),
    Pessoa('Joao', 34),
    Pessoa('Joaozinho', 0),
]

print "Maiores de idade:"

for p in pessoas:
    if (p.maior_de_idade()):
        print p.nome()
